File: app/src/main/java/ch/abler/aline/alarmdroid/syncadapter/AlarmSyncService.java
package ch.abler.aline.alarmdroid.syncadapter;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.support.annotation.Nullable;

public class AlarmSyncService extends Service {

    private static AlarmSyncAdapter syncAdapter = null;
    private static final Object syncAdapterLock = new Object();

    @Override
    public void onCreate() {
        super.onCreate();

        synchronized (syncAdapterLock) {
            if (syncAdapter == null) {
                syncAdapter = new AlarmSyncAdapter(getApplicationContext(), true);
            }
        }
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return syncAdapter.getSyncAdapterBinder();
    }
}

File: app/src/main/java/ch/abler/aline/alarmdroid/sync/AlarmSyncService.kt
package ch.abler.aline.alarmdroid.sync

import android.app.Service
import android.content.Intent
import android.os.IBinder

class AlarmSyncService : Service() {
    override fun onCreate() {
        synchronized(syncAdapterLock) {
            syncAdapter = syncAdapter ?: AlarmSyncAdapter(applicationContext, true)
        }
    }

    override fun onBind(intent: Intent?): IBinder? {
        return syncAdapter!!.syncAdapterBinder
    }

    companion object {
        private var syncAdapter: AlarmSyncAdapter? = null
        private val syncAdapterLock = Any()
    }
}
